### PR TITLE
Restrict payment information

### PR DESCRIPTION
**Pull request by nicolasnunez-eb**
Jira ticket : EDA-623
Now some fields ("Payment term", "Payment type" and "Bank name") are restricted to default values.

### CODE REVIEW

File: supplier_app/migrations/0001_initial.py [hunk 1]
```diff
@@ -31,9 +33,8 @@ class Migration(migrations.Migration):
             name='BankAccount',
             fields=[
                 ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
-                ('bank_name', models.CharField(default='', max_length=200)),
-                ('bank_code', models.CharField(default='', max_length=200)),
-                ('bank_account_number', models.CharField(default='', max_length=200)),
+                ('bank_account_number', models.CharField(max_length=60, unique=True)),
+                ('bank_info', models.IntegerField(choices=[(16, 'CITIBANK N.A.'), (7, 'BANCO DE GALICIA Y BUENOS AIRES S.A.'), (11, 'BANCO DE LA NACION ARGENTINA'), (17, 'BBVA BANCO FRANCES S.A.')], verbose_name='Bank name')),
```
Comment (leandro-eb) [inline supplier_app/migrations/0001_initial.py]: Should we use a tuple here?

File: supplier_app/migrations/0001_initial.py [hunk 2]
```diff
@@ -66,11 +68,12 @@ class Migration(migrations.Migration):
             name='TaxPayerArgentina',
             fields=[
                 ('taxpayer_ptr', models.OneToOneField(auto_created=True, on_delete=django.db.models.deletion.CASCADE, parent_link=True, primary_key=True, serialize=False, to='supplier_app.TaxPayer')),
-                ('cuit', models.CharField(max_length=200)),
-                ('comments', models.CharField(max_length=200)),
-                ('payment_type', models.CharField(max_length=200)),
+                ('cuit', models.CharField(max_length=20)),
+                ('payment_type', models.CharField(choices=[('BANK', 'Bank')], max_length=20)),
```
Comment (leandro-eb) [inline supplier_app/migrations/0001_initial.py]: I'm wondering if 'max_length' is still necessary or not

File: supplier_app/migrations/0001_initial.py [hunk 3]
```diff
@@ -66,11 +68,12 @@ class Migration(migrations.Migration):
...
+                ('payment_term', models.IntegerField(choices=[(15, '15 days'), (30, '30 days')], default=None)),
```
Comment (leandro-eb) [inline supplier_app/migrations/0001_initial.py]: The default value should be `None` or should we ask our PM?

File: supplier_app/models.py [hunk 4]
```diff
@@ -73,6 +84,11 @@ class TaxPayerArgentina(TaxPayer):
         verbose_name='Certificate of no tax withholding of IVA, income or SUSS',
         validators=[FileExtensionValidator(allowed_extensions=['pdf'])],
         )
+    comments = models.CharField(
+        max_length=200,
+        blank=True,
+        null=True
+        )
```
Comment (leandro-eb) [inline supplier_app/models.py]: `comments` field is only available for `TaxPayerArgentina` or should be generic?

File: supplier_app/tests/factory_boy.py [hunk 5]
```diff
@@ -49,6 +49,7 @@ class Meta:
     cuit = "1234569"
     comments = "Justificacion falsa"
     payment_type = "Banco"
```
Comment (leandro-eb) [inline supplier_app/tests/factory_boy.py]: `payment_type` should be one of a list of possible payment type values

File: supplier_app/tests/factory_boy.py [hunk 6]
```diff
@@ -49,6 +49,7 @@ class Meta:
...
+    payment_term = 15
```
Comment (leandro-eb) [inline supplier_app/tests/factory_boy.py]: `payment_term` should be one of a list of possible payment term values

File: supplier_app/tests/test_view.py [hunk 7]
```diff
@@ -64,6 +65,7 @@ def setUp(self):
         self.file_mock = MagicMock(spec=File)
         self.file_mock.name = 'test.pdf'
         self.file_mock.size = 50
+        self.bank_info = BANK_INFO["CITIBANK N.A."]
```
Comment (leandro-eb) [inline supplier_app/tests/test_view.py]: Please use a Factory for bank_info

File: supplier_app/tests/test_view.py [hunk 8]
```diff
@@ -520,155 +526,97 @@ def setUp(self):
...
+        self.kwargs_taxpayer = {
             'taxpayer_id': self.taxpayer.id,
         }
+        self.kwargs_address = {
+            'address_id': self.address.id
+        }
         self.taxpayer_id = self.taxpayer.id
 
-    def test_get_success_url_should_redirect_to_details_view_when_click_in_update_button(self):
+    def test_get_detail_view(self):
```
Comment (leandro-eb) [inline supplier_app/tests/test_view.py]: Please provide context of what are you testing on this specific test on the name, or add a comment inside

File: supplier_app/models.py (hunk 4 above)
Comment (nicolasnunez-eb) [reply supplier_app/models.py]: Resolved, comments is now available for every taxpayer

File: supplier_app/tests/factory_boy.py (hunk 5 above)
Comment (nicolasnunez-eb) [reply supplier_app/tests/factory_boy.py]: resolved

File: supplier_app/tests/factory_boy.py (hunk 6 above)
Comment (nicolasnunez-eb) [reply supplier_app/tests/factory_boy.py]: resolved

File: supplier_app/tests/test_view.py (hunk 8 above)
Comment (nicolasnunez-eb) [reply supplier_app/tests/test_view.py]: resolved

File: supplier_app/tests/test_view.py (hunk 7 above)
Comment (nicolasnunez-eb) [reply supplier_app/tests/test_view.py]: resolved

File: supplier_app/migrations/0001_initial.py (hunk 3 above)
Comment (nicolasnunez-eb) [reply supplier_app/migrations/0001_initial.py]: For now the default value is 15 days, so I change that

File: supplier_app/migrations/0001_initial.py (hunk 2 above)
Comment (nicolasnunez-eb) [reply supplier_app/migrations/0001_initial.py]: Django ORM demands you to put a max_length

File: supplier_app/migrations/0001_initial.py (hunk 1 above)
Comment (nicolasnunez-eb) [reply supplier_app/migrations/0001_initial.py]: That code is made by the ./manage.py makemigrations command and we are not supposed to modify these files

File: supplier_app/tests/__init__.py [hunk 9]
```diff
@@ -0,0 +1,56 @@
+from supplier_app import (
+    PAYMENT_TERMS,
+    PAYMENT_TYPES,
+)
+from supplier_app.bank_info import BANK_INFO
+
+
+def get_bank_info_example(key=None):
+    return BANK_INFO[key or "BANCO DE LA NACION ARGENTINA"]
```
Comment (leandro-eb) [inline supplier_app/tests/__init__.py]: I'm wondering where are we using this function

Comment (nicolasnunez-eb) [reply supplier_app/tests/__init__.py]: I use it "TestCreateTaxPayer" when I create a bank_info

**Pull request merged**